packages/wdio-config/src/dataproviders.js

```diff
@@ -0,0 +1,42 @@
+import { safeRequire } from '@wdio/utils'
+import ConfigParser from './lib/ConfigParser'
+
+/**
+ * Loads the data providers from the files to memory
+ * @param  {Object} dataProviderFilesPaths  List of files with data provider injections
+ */
+export function initializeDataProviders (dataProviderFilesPaths) {
+    const dataProviderFiles =  ConfigParser.getFilePaths(dataProviderFilesPaths)
+    const dataProvidersMap = {}
+    let currentDataProviderFile = ''
+
+    global.dataProvider = function (specFilePath, dataSet) {
```

Comment:
Don't understand why you're attaching this to the global object and then deleting it later in the block. Does it have to be like this?